src/query/pipeline/sources/src/input_formats/impls/input_format_parquet.rs

```diff
@@ -489,3 +506,27 @@ pub fn split_column_metas_by_field(
     });
     r
 }
+
+fn get_field_columns<'a>(
+    columns: &'a [ColumnChunkMetaData],
+    field_name: &str,
+) -> Vec<&'a ColumnChunkMetaData> {
+    columns
+        .iter()
+        .filter(|x| x.descriptor().path_in_schema[0] == field_name)
+        .collect()
+}
+
+async fn _read_single_column_async<R>(
```

Comment:
Why does this function's name starts with underline?

Reply:
done